### ISSUE TITLE
buggy strategy for setting a11y name of radio button group

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/sun#511]
Noticed while working on https://github.com/phetsims/sun/issues/510...

`VerticalAquaRadioButtonGroup` and `RadioButtonGroup` both have this:

```js
  // a11y - An id for each instance of VerticalAquaRadioButtonGroup, passed to individual buttons in the group.
  // Each button in a radio button group must have the same "name" attribute to be considered in a group, otherwise
  // arrow keys will navigate all radio type inputs in the document.
  var instanceCount = 0;
```

`VerticalAquaRadioButtonGroup` does:

```js
      var radioButton = new AquaRadioButton( property, item.value, content,
        _.extend( {}, options.radioButtonOptions, {
          tandem: item.tandemName ? options.tandem.createTandem( item.tandemName ) : Tandem.required,
          labelContent: item.labelContent || null,
          a11yNameAttribute: instanceCount
        } ) );
```

RadioButtonGroup does:
```js
      // a11y - so the browser recognizes these buttons are in the same group, see instanceCount for more info
      radioButton.setAccessibleAttribute( 'name', instanceCount );
```

This looks like trouble. I discussed with @jessegreenberg and he agreed.  If you have both a `VerticalAquaRadioButtonGroup` and a `RadioButtonGroup` in a sim, arrow keys will likely navigate directly between them.  

If having a "group" is desired, then that should be factored out and used for both Aqua and rectangular buttons.

**Comment by pixelzoom** [comment]
The other way that this could be resolved is to add a unique prefix to `instanceCount`.  That might be the best approach, since we might create other "group" UI components in the future, either in common code or in sims.  Recommended to use the full name of the class as the prefix, e.g. `"VerticalAquaRadioButtonGroup1"`.

**Comment by jessegreenberg** [comment]
> arrow keys will likely navigate directly between them.

Our prediction was correct, arrow keys currently loop through both the AquaRadioButtonGroup and the RadioButtonGroup in the sun ButtonsScreenView demo.

**Comment by jessegreenberg** [comment]
I added a prefix to the instanceCount with the class name and tested to make sure the bug was fixed in sun ButtonsScreenView demo. @pixelzoom does this seem OK?

**Comment by pixelzoom** [comment]
It's unfortunate that we have to duplicate the class name again as a string literal.  But I don't see any way around that, since constructor/class names get mangled when a sim is built. 

Closing.
